### PR TITLE
Implemented enum Colors, class ColorSupplier with getRandomColor meth…

### PR DESCRIPTION
**Pull request by DrTymkivstain**
…od, class Ball with overriding toString method, clas Lottery that creates a random ball using implemented classes, tests in class Application

### CODE REVIEW

**Review by KostyaNikiforov — CHANGES_REQUESTED** [review]
Check the checklist one more time please) There is a point about static methods and constants
And also check my message in the group chat about dependencies

**Comment by DrTymkivstain** [comment]
> Check the checklist one more time please) There is a point about static methods and constants And also check my message in the group chat about dependen

**Comment by DrTymkivstain** [comment]
> Check the checklist one more time please) There is a point about static methods and constants And also check my message in the group chat about dependencies

I will try to. Do I need to remove the setters from the ball class and make the fields public? or how do I work with them in the method getRandomBall() ?

**Comment by DrTymkivstain** [comment]
> Check the checklist one more time please) There is a point about static methods and constants And also check my message in the group chat about dependencies

and  do I need to create a constructor in the ColorSupplier class and pass it in random from main class?

File: src/main/java/core/basesyntax/Lottery.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+    public Lottery(Random random, ColorSupplier colorSupplier) {
+        this.random = random;
+        this.colorSupplier = colorSupplier;
+    }
+
+    public Ball getRandomBall() {
+        Ball randomBall = new Ball();
+        randomBall.setColor(colorSupplier.getRandomColor());
+        randomBall.setNumber(random.nextInt(100));
```
Comment (KostyaNikiforov) [inline src/main/java/core/basesyntax/Lottery.java:17]: Let's make the value 100 constant, as the checklist requires

**Review by KostyaNikiforov — CHANGES_REQUESTED** [review]
Exactly!)

**Review by KostyaNikiforov — APPROVED** [review]
🔥

**Pull request closed without merging**